### ISSUE TITLE
Argument order of "Api::api" method is not correct

### ISSUE THREAD

**Issue by dereuromark** [console-helpers/jira-api-restclient#116]
The order as per clean code should not have optional values prior to required ones:

``` php
    public function api(
        $method = self::REQUEST_GET,
        $url,
        $data = array(),
        $return_as_json = false,
        $is_file = false,
        $debug = false
    )
```

Instead if should be

``` php
    public function api(
        $url,
        $method = self::REQUEST_GET,
        $data = array(),
        $return_as_json = false,
        $is_file = false,
        $debug = false
    )
```

So you can actually do

``` php
    ->api($url)
```

and the others are actually optional.

**Comment by aik099** [comment]
Yeah, I know. This is the problem with several other methods as well. PhpStorm always complains about it to me.

Unfortunately we can't change argument order because that would be a BC break.

**Comment by aik099** [comment]
What we can do is to remove default value for `$method` argument. That sound like BC break, but in fact it is not, because when manually calling `Api::api` method you have to specify `$method` all the time and trick like specifying `null` instead to get default value used won't work currently.

**Comment by dereuromark** [comment]
I agree.
And maybe in a future major bump we can fix it.

**Comment by aik099** [comment]
So, if you'd like you can send PR that would remove default value for optional method arguments, that come (in method declaration) before non-optional arguments. There are several such methods for sure.

**Comment by aik099** [comment]
> And maybe in a future major bump we can fix it.

I don't think, that changing argument order is a way to go, because in API calls the result changes greatly based on the request method (get/post/etc) so specifying it first and then url makes sense actually.

**Comment by aik099** [comment]
@dereuromark , should I expect PR for review soon?

**Comment by aik099** [comment]
Closing in favor of #226.
